stdlib/toolchain/CompatibilityBytecodeLayouts/BytecodeLayouts.h

```diff
@@ -0,0 +1,301 @@
+//===--- RuntimeValueWitness.h                                         ---===//
+// Swift Language Bytecode Layouts Runtime Implementation
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Implementations of runtime determined value witness functions
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_BYTECODE_LAYOUTS_H
+#define SWIFT_BYTECODE_LAYOUTS_H
+
+#include "swift/Runtime/Metadata.h"
+#include <cstdint>
+#include <vector>
+
+// Layouts
+//
+// enum class LayoutType: char {
+//   // Scalars
+//   I8 = 'c',
```

Comment:
Missing (at least) i1.

Reply:
I don't think the scalar encoding as is is sufficient. LLVM builtin types are available -- eventually, we need to be able to match/encode the results of `getSpareBitsForType`.